internal/publish/pub_test.go

```diff
@@ -106,6 +104,8 @@ func TestPublisherStopShutdownInactive(t *testing.T) {
 }
 
 func BenchmarkPublisher(b *testing.B) {
```

Comment:
Do you know why the benchmark is hanging? I think we need to know why before disabling it, in case there's some real issue being masked.